lib/text/vtt_text_parser.js

```diff
@@ -219,6 +221,46 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {!shaka.text.Cue} cue
+   * @param {string} payload
+   */
+  static parseCueStyles(cue, payload) {
+    try {
+      const parser = new DOMParser();
+      const xml = parser.parseFromString(payload, 'text/xml');
+      const parserError = xml.getElementsByTagName('parsererror')[0];
+      if (parserError) {
+        throw new shaka.util.Error(
+            shaka.util.Error.Severity.CRITICAL,
+            shaka.util.Error.Category.TEXT,
+            shaka.util.Error.Code.INVALID_XML,
+            parserError.textContent);
+      }
+      const element = xml.documentElement;
+      // TODO: Add support for nested elements at more than two levels of nesting
```

Comment:
I think of it as a requirement that we parse nested tags here if we parse any at all.